internal/migration/precheck.go

```diff
@@ -426,6 +433,18 @@ type agentToolsGetter interface {
 	AgentTools() (*tools.Tools, error)
 }
 
+// checkNoFanConfig makes sure that no fan config was used in the config of the
+// model being migrated.
+func checkNoFanConfig(modelConfig map[string]interface{}) error {
+	if modelConfig[config.FanConfig] != nil && modelConfig[config.FanConfig] != "" {
+		return errors.Errorf("fan networking not supported, fan config %q", modelConfig[config.FanConfig])
+	}
+	if modelConfig[config.ContainerNetworkingMethod] != nil && modelConfig[config.ContainerNetworkingMethod] == "fan" {
+		return errors.Errorf("fan networking not supported, container networking method %q", modelConfig[config.ContainerNetworkingMethod])
+	}
```

Comment:
I think we should be more descriptive with the error message.

Something like:

> fan networking not supported, remove fan config %q from migrating model config

Reply:
I agree, fixed.